Lib/test/support/script_helper.py

```diff
@@ -84,6 +85,21 @@ def fail(self, cmd_line):
                                 err))
 
 
+def run_until(condition, timeout=60, error_msg=None):
+    """Evaluates a condition until it becomes True, or a timeout is reached."""
```

Comment:
I don't understand how I am supposed to use to ... run something.

Reply:
I'm sorry, I did not understand your comment. Can you re-phrase it?

Reply:
Example: https://bugs.python.org/issue36950#msg342751

Why do I put this code?

```
                try:
                    smm = shared_memory.SharedMemory(name, create=False)
                except FileNotFoundError:
                    break
```

Reply:
This is what is explained [here](https://bugs.python.org/issue36950#msg342804).
From what I saw in the `stdlib` tests, there are not so many cases that are that complex. 
Many tests use much simpler conditions (usually, they check the result of a function)

Example: in `_test_multiprocessing.py`:
```python
    def wait_proc_exit(self):
        # Only the manager process should be returned by active_children()
        # but this can take a bit on slow machines, so wait a few seconds
        # if there are other children too (see #17395).
        join_process(self.proc)
        start_time = time.monotonic()
        t = 0.01
        while len(multiprocessing.active_children()) > 1:
            time.sleep(t)
            t *= 2
            dt = time.monotonic() - start_time
            if dt >= 5.0:
                test.support.environment_altered = True
                print("Warning -- multiprocessing.Manager still has %s active "
                      "children after %s seconds"
                      % (multiprocessing.active_children(), dt),
                      file=sys.stderr)
                break
```

We could change this test into 
```python
def wait_proc_exit(self):
    join_process(self.proc)
    run_until(lambda: len(multiprocessing.active_children()) == 1, timeout=5)
```

You can see a exhaustive list of the tests that can be re-written using the current implementation of `run_until`  here: https://bugs.python.org/issue36950#msg342751. As I said, I did not find so many more complex test cases such as the one you are pointing out.

Knowing this, do we still want a more powerful `run_until`?